### PR TITLE
Reuse internal values array in StorePropertyCursor

### PR DESCRIPTION
**Pull request by davidegrohmann**
This should help in reducing GC.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java [hunk 1]
```diff
@@ -78,14 +80,15 @@
...
     private PropertyType type;
     private int keyId;
     private ByteBuffer buffer;
 
     private AbstractDynamicStore.DynamicRecordCursor recordCursor;
     private long currentRecordId;
-    private long[] valuesAfterHeader;
+
+    // in values[0] will be placed the property block header
+    private long[] values = new long[VALUES_SIZE];
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java]: Why not to split those 2 things then? variable for value ( current value[0]?) and all the rest for remaining array?

Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java]: it is easier this way when you need to copy all those data into a byte buffer, to make it nicer and avoid confusion we can have a small method called `header()` which will give you the right data from that array. WDYT?

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java]: that will a bit improve value[0], but value[1] still be there :)

what about wrapping `values` into helper class and provide `header()`, `toBits()`, `remainingHead()` and `setRemaining(index)` or similar there. That will keep a way how data can be copied into buffer and will hide from `StorePropertyCursor` all array indexing juggling? WDYT?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java [hunk 2]
```diff
@@ -573,4 +510,100 @@ private void assertOfOneOfTypes( PropertyType type1, PropertyType type2 )
...
+        public static final int VALUES_SIZE = DEFAULT_PAYLOAD_SIZE / 8;
+
+        private int writeIndex;
+        private long[] values = new long[VALUES_SIZE];
+        private int remaining;
+
+        public void init()
+        {
+            writeIndex = 0;
+            remaining = 0;
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java:525]: lets clear values[] here as well WDYT?

**Comment by MishaDemianenko** [comment]
@davidegrohmann small comment all the rest is fine

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java (hunk 2 above)
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java:525]: I can do that, but what is the benefit? I think I made sure we read only something that has been written, no?

Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java:525]: yes that's true indexes now looks fine, just to be safe (also for future modifications) and sure that in any case we can't read anything obsolete from `Block`.

Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java:525]: Done

**Pull request merged**